services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -594,24 +650,151 @@ testGetTeamMembersIncludesLHStatus = do
-  check UserLegalHoldDisabled "disabled when it is disabled for the team"
+
+  check UserLegalHoldNoConsent "disabled when it is disabled for the team"
   withDummyTestServiceForTeam owner tid $ \_chan -> do
-    check UserLegalHoldDisabled "disabled on new team members"
+    check UserLegalHoldNoConsent "no_consent on new team members"
+    grantConsent tid member
+    check UserLegalHoldDisabled "disabled on team members that have granted consent"
     requestLegalHoldDevice owner member tid !!! testResponse 201 Nothing
     check UserLegalHoldPending "pending after requesting device"
     approveLegalHoldDevice (Just defPassword) member member tid !!! testResponse 200 Nothing
     check UserLegalHoldEnabled "enabled after confirming device"
 
+testNotInWhitelist :: TestM ()
+testNotInWhitelist = do
+  g <- view tsGalley
+  (owner, tid) <- createBindingTeam
+  member <- randomUser
+  addTeamMemberInternal tid member (rolePermissions RoleMember) Nothing
+  ensureQueueEmpty
+  (_, tid') <- createBindingTeam -- we enable for this team, and then see how the above team behaves.
+  cannon <- view tsCannon
+  WS.bracketR2 cannon member member $ \(_ws, _ws') -> withDummyTestServiceForTeam owner tid $ \_chan -> do
+    do
+      -- enable NOT-whitelisted team should fail
+      withLHWhitelist tid' (putEnabledM' g id tid Public.TeamFeatureEnabled)
+        !!! testResponse 403 (Just "legalhold-whitelisted-only")
+
+      -- disable whitelisted team should fail
+      withLHWhitelist tid (putEnabledM' g id tid Public.TeamFeatureDisabled)
+        !!! testResponse 403 (Just "legalhold-whitelisted-only")
+
+      -- enable whitelisted team should fail (no need for this to work)
+      withLHWhitelist tid (putEnabledM' g id tid Public.TeamFeatureEnabled)
+        !!! testResponse 403 (Just "legalhold-whitelisted-only")
+
+      -- disable NOT-whitelisted team should fail (no need for this to work)
+      withLHWhitelist tid' (putEnabledM' g id tid Public.TeamFeatureDisabled)
+        !!! testResponse 403 (Just "legalhold-whitelisted-only")
+
+    do
+      -- members have not granted consent implicitly...
+      lhs <- view legalHoldStatus <$> withLHWhitelist tid' (getTeamMember' g member tid member)
+      liftIO $ assertEqual "" lhs UserLegalHoldNoConsent
+
+      -- ... but can do so explicitly.
+      withLHWhitelist tid' (grantConsent' g tid member)
+      lhs' <- withLHWhitelist tid' $ view legalHoldStatus <$> getTeamMember' g member tid member
+      liftIO $ assertEqual "" lhs' UserLegalHoldDisabled
+
+    do
+      -- nobody can request LH device.
+      withLHWhitelist tid' (requestLegalHoldDevice' g owner member tid) !!! testResponse 403 (Just "legalhold-not-enabled")
+      UserLegalHoldStatusResponse userStatus _ _ <- withLHWhitelist tid' (getUserStatusTyped' g member tid)
+      liftIO $
+        assertEqual
+          "requestLegalHoldDevice should set user status to Pending"
+          UserLegalHoldDisabled
```

Comment:
The string in assertEqual doesn't seem to reflect what is being tested here. Equality is tested for Disabled, not Pending. Am I holding this wrong or could this be made clearer / the text changed?

Reply:
copy&paste error, thanks!  35f487e80563706f5a60532161683a671405213c